src/main/scala/com/gu/salesfoce/messageHandler/MessageHandlerLambda.scala

```diff
@@ -0,0 +1,111 @@
+      |		</notificationsResponse>
+      |	</soapenv:Body>
+      |</soapenv:Envelope>
+    """.stripMargin
+
+  val okResponse = ApiResponse("200", Headers(), okXml)
+
+  def parseMessage(requestBody: String): List[ContactNotification] = {
+    val is = new ByteArrayInputStream(requestBody.getBytes)
+    val messageFactory = MessageFactory.newInstance()
+    val soapMessage = messageFactory.createMessage(null, is)
+    val body = soapMessage.getSOAPBody
+    val jc = JAXBContext.newInstance(classOf[Notifications])
+    val unmarshaller = jc.createUnmarshaller()
+    val je = unmarshaller.unmarshal(body.extractContentAsDocument(), classOf[Notifications])
+    val notifications = je.getValue()
+    notifications.getNotification.toList
+  }
+
+  case class QueueMessage(contactId: String)
+
+  implicit val messageFormat = Json.format[QueueMessage]
+
+  def credentialsAreValid(inputEvent: JsValue): Boolean = {
+
+    val maybeApiClientId = (inputEvent \ "queryStringParameters" \ "apiClientId").asOpt[String]
+    val maybeApiClientToken = (inputEvent \ "queryStringParameters" \ "apiToken").asOpt[String]
+    val maybeCredentials = (maybeApiClientId, maybeApiClientToken)
+    maybeCredentials match {
+      case (Some(apiClientId), Some(apiToken)) => {
+        (apiClientId == Config.apiClientId && apiToken == Config.apiToken)
+      }
+      case _ => {
+        logger.info(s"Could not find credentials in request")
+        false
+      }
+    }
+  }
+
+  def sendToQueue(notification: ContactNotification): Future[Try[SendMessageResult]] = {
+    val queueMessage = QueueMessage(notification.getSObject.getId)
+    val queueMessageString = Json.prettyPrint(Json.toJson(queueMessage))
+    queueClient.send(queueName, queueMessageString)
+  }
+
+  def handleRequest(inputStream: InputStream, outputStream: OutputStream, context: Context): Unit = {
+
+    logger.info(s"Salesforce message handler lambda ${Config.stage} is starting up...")
+    val inputEvent = Json.parse(inputStream)
+    if (!credentialsAreValid(inputEvent)) {
+      logger.info("Request from Zuora could not be authenticated")
+      outputForAPIGateway(outputStream, unauthorized)
+    } else {
+      logger.info("Authenticated request successfully...")
+      val body = (inputEvent \ "body").as[String]
+      val notifications = parseMessage(body)
+      logger.info(s"found ${notifications.size} contact(s) in the salesforce xml")
+      val FutureResponses = notifications.map(sendToQueue)
+      val future = Future.sequence(FutureResponses).map { responses =>
+        val errors = responses collect { case Failure(error) => error }
+        if (errors.nonEmpty) {
+          errors.foreach(error =>
+            logger.error(s"error while trying to send message to queue", error))
+          outputForAPIGateway(outputStream, internalServerError)
+        } else {
+          outputForAPIGateway(outputStream, okResponse)
+        }
+      }
+      Await.ready(future, Duration.Inf)
```

Comment:
that is the official way to wait for a future, but there should be no reason for anyone to need to do that unless they are the very outer layer of the code i.e the framework that kicks everything off. If you were using play, the controller would return a future to Action.async and return the action. In this case, you are the outer layer.
Using a synchronous client would also work, but wouldn't conform to pushing the side effects to the edge, so it depends how far you want to go with that.